azure-keyvault-cryptography/src/main/java/com/microsoft/azure/keyvault/cryptography/ByteExtensions.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.keyvault.cryptography;
+
+public final class ByteExtensions {
+
+    public static boolean sequenceEqualConstantTime( byte[] self, byte[] other )
+    {
+        if ( self == null )
+            throw new IllegalArgumentException( "self" );
+
+        if ( other == null )
+            throw new IllegalArgumentException( "other" );
+
+        // Constant time comparison of two byte arrays
+        long difference = ( self.length & 0xffffffffl ) ^ ( other.length & 0xffffffffl );
+
+        for ( int i = 0; i < self.length && i < other.length; i++ )
+        {
+            difference |= ( self[i] ^ other[i] ) & 0xffffffffl;
+        }
+
+        return difference == 0;
+    }
+
+    public static byte[] or( byte[] self, byte[] other )
+    {
+        return or( self, other, 0 );
+    }
+
+    public static byte[] or( byte[] self, byte[] other, int offset )
+    {
+        if ( self == null )
+            throw new IllegalArgumentException( "self" );
+
+        if ( other == null )
+            throw new IllegalArgumentException( "other" );
+
+        if ( self.length > other.length - offset )
+            throw new IllegalArgumentException( "self and other lengths do not match" );
+
+        byte[] result = new byte[self.length];
+
+        for ( int i = 0; i < self.length; i++ )
+        {
+            result[i] = (byte)( self[i] | other[offset + i] );
+        }
+
+        return result;
+    }
+    
+    public static byte[] xor( byte[] self, byte[] other ) {
+    	return xor( self, other, 0, false );
+    }
+
+    public static byte[] xor( byte[] self, byte[] other, boolean inPlace )
+    {
+        return xor( self, other, 0, inPlace );
+    }
+    
+    public static byte[] xor( byte[] self, byte[] other, int offset ) {
+    	return xor( self, other, 0, false );
+    }
+
+    public static byte[] xor( byte[] self, byte[] other, int offset, boolean inPlace )
```

Comment:
This boolean flag is not good design. It would be best to take a third array:

public static void xor( byte[] self, byte[] other, int offset, byte[] result ).

If the result is not null, it's used. The user can specify the same value it uses for self or other, or a third array.

If the result is null, it's created.